### PR TITLE
fix(ProtoBuf,XML): component data repeatable

### PR DESCRIPTION
**Pull request by jkowalleck**
fixes <https://github.com/CycloneDX/specification/issues/518>

### CODE REVIEW

**Comment by jkowalleck** [comment]
@nscuro , could I ask you for a review?

would you consider the proposed changes as non-breaking? 
I got the impression this would be a non-breaking based on this: https://github.com/CycloneDX/specification/issues/518#issuecomment-2334405333 

Despite `bufbuild/buf:1.30.1` thinks this would be a breaking change ...

File: tools/src/test/resources/1.6/valid-component-data-1.6.textproto [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+        name: "app.data"
+        contents {
+            url: "https://example.com/cfg/1337/app.ini"
+        }
+    }
+    data {
+        type: COMPONENT_DATA_TYPE_OTHER
+        name: ".env"
+        contents {
+            url: "https://example.com/cfg/1337/dotenv"
```
Comment (andreas-hilti) [inline tools/src/test/resources/1.6/valid-component-data-1.6.textproto:22]:
```suggestion
            url: "https://example.com/cfg/1337/env"
```
for consistency?

**Review by andreas-hilti — APPROVED** [review]

**Comment by nscuro** [comment]
@jkowalleck You are using the config category `FILE` for `buf`'s breaking change detection:

https://github.com/CycloneDX/specification/blob/19a153072690dfb2e8475ea3fa2e09a657d0cef6/tools/src/test/proto/buf_breaking-version.yaml#L3-L7

The [docs](https://buf.build/docs/breaking/overview/#rules-and-categories) define `FILE` as:

> Detects generated source code breakage on a per-file basis

So, while the proposed change is compatible WRT the wire format, source code generated from the schema will break. For example in Java, `optional ComponentData data` would be accessible as:

```java
// Returns ComponentData
component.getData()
```

whereas `repeated ComponentData data` would be:

```java
// Returns List<ComponentData>
component.getDataList()
```

Similarly, *construction* of Proto objects will be different, too.

It's safe to assume that almost all language bindings will mirror this behavior.

IMO this is fine. I think the spec should focus on wire compatibility.

**Comment by jkowalleck** [comment]
thank you so much, @nscuro.

I am not very experienced with ProtoBuf, so I configured the checks in an unideal way.

> IMO this is fine. I think the spec should focus on wire compatibility. 

Would you say it is safe to change the breaking detection to use "wire" only, by dropping "file"? 
If so, I will change the applied configs

**Review by stevespringett — APPROVED** [review]

**Comment by stevespringett** [comment]
LGTM

**Pull request merged**